### ISSUE TITLE
Examine additional errors in SQL exceptions to properly handle connections

### ISSUE THREAD

**Issue by JeremyLikness** [dotnet/efcore#27917]
Specifically look at error number [18456](https://docs.microsoft.com/sql/relational-databases/errors-events/mssqlserver-18456-database-engine-error?view=sql-server-ver15).

Related to issue [#15644](https://github.com/dotnet/efcore/issues/15644) and PR [25832](https://github.com/dotnet/efcore/pull/25832/).

**Comment by AndriySvyryd** [comment]
> Is it save to add this code to additional transient errors that should be retried?

Yes, the worst that can happen is that the application takes 30 seconds longer to throw an exception

**Comment by edowns-btown** [comment]
It seems like when using managed identity in Azure, attempting to connect to a database that does NOT exist will always result in an 18456, "Login failed for user '<token-identified principal>' ".    Throwing in that case may not be the correct thing to do.

Using the master database to check for existence may be the way to go.   I had to override SqlServerDatabaseCreator.ExistsAsync to do a check with master in order to get migrations to work correctly when the database does not yet exist.

**Comment by pregress** [comment]
The same error happens when performing scaling operations and connecting with managed identity. 
I feel like this should be handle here: https://github.com/dotnet/efcore/blob/20edb631265527a73cb9b9ab474cd0484c49a771/src/EFCore.SqlServer/SqlServerRetryingExecutionStrategy.cs#L221

**Comment by Rookian** [comment]
Shouldn't this error be already added when using `UseAzureSql`?
